org.eclipse.jdt.debug/model/org/eclipse/jdt/internal/debug/core/breakpoints/ValidBreakpointLocationLocator.java

```diff
@@ -1066,10 +1066,16 @@ public boolean visit(LambdaExpression node) {
 
 						}
 					}
+
+				// Lambda body can be a block (which is handled above) or an (arbitrary) expression. So maybe the
+				// following cases are insufficient and should be replaced by handling the general
```

Comment:
sounds odd to know what todo and not doing it ;-) could you instead of a comment suggest such an appropriate fix?

Reply:
Well, I am not 100% sure if the extended version of the fix would be OK. Maybe the gurus for Java grammar and AST can give a short advice here. @stephan-herrmann ?  But what is clear according to my tests is that the fix "as is" solves the described bug.

Reply:
Right, just handling the two specified alternatives Block and Expression sounds like the most consistent approach.

Seeing other visit methods freely invoking `.accept(this)` on general expressions, I see no reason why this should cause problems here, but I haven't studied class ValidBreakpointLocationLocator in more detail.